__tests__/endpoints.test.js

```diff
@@ -180,4 +180,84 @@ describe("/api/reviews/:review_id/comments", () => {
         expect(response.body.message).toBe("Review not found :(");
       });
   });
+  test("GET - status 400, responds with error message when passed review number is invalid", () => {
+    return request(app)
+      .get("/api/reviews/something/comments")
+      .expect(400)
+      .then((response) => {
+        expect(response.body.message).toBe("Bad request :(");
+      });
+  });
+});
+describe("POST, /api/reviews/:review_id/comments", () => {
+  test("POST - status 201 - responds with fresh-posted comment", () => {
+    const newComment = {
+      username: "dav3rid",
+      body: "OMG this game is so good I spent 2 days with no rest playing it",
+    };
+    return request(app)
+      .post("/api/reviews/5/comments")
+      .send(newComment)
+      .expect(201)
+      .then((response) => {
+        const comment = response.body.comment;
+        expect(comment.review_id).toBe(5);
+        expect(comment.author).toBe("dav3rid");
+        expect(comment.body).toBe(
+          "OMG this game is so good I spent 2 days with no rest playing it"
+        );
+        expect(comment.votes).toBe(0);
+      });
+  });
+  test("POST - status 400 - responds with error message when one or more fields of comment object missing", () => {
+    const newComment = {
+      username: "dav3rid",
+    };
+    return request(app)
+      .post("/api/reviews/5/comments")
+      .send(newComment)
+      .expect(400)
+      .then((response) => {
+        expect(response.body.message).toBe("Missing necessary property!");
+      });
+  });
+  test("POST - status 400 - responds with error message when one or more fields of comment object have incorrect type", () => {
+    const newComment = {
+      username: "dav3rid",
+      body: 8,
+    };
+    return request(app)
+      .post("/api/reviews/5/comments")
+      .send(newComment)
+      .expect(400)
+      .then((response) => {
+        expect(response.body.message).toBe("Incorrect property type!");
+      });
+  });
+  test("POST - status 404, responds with error message when object with passed review number doesn't exist", () => {
+    const newComment = {
+      username: "dav3rid",
+      body: "OMG this game is so good I spent 2 days with no rest playing it",
+    };
+    return request(app)
+      .post("/api/reviews/88888888/comments")
+      .send(newComment)
+      .expect(404)
+      .then((response) => {
+        expect(response.body.message).toBe("Review not found :(");
+      });
+  });
+  test("POST - status 400, responds with error message when passed review number is invalid", () => {
+    const newComment = {
+      username: "dav3rid",
+      body: "OMG this game is so good I spent 2 days with no rest playing it",
+    };
+    return request(app)
+      .post("/api/reviews/something/comments")
+      .send(newComment)
+      .expect(400)
+      .then((response) => {
+        expect(response.body.message).toBe("Bad request :(");
+      });
+  });
```

Comment:
What if a post is made with a non existent username? What test could we add for this 💭